vignettes/rationale.Rmd

```diff
@@ -0,0 +1,128 @@
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+This document explains the rationale behind the development of this
+algorithm. Many of these text were taken from Anders Aasted Isaksen's
+[PhD Thesis](https://aastedet.github.io/dissertation/) as well as the
+validation paper [@Isaksen2023]. This document is a shorter and more
+concise version of those documents. We cover the:
+
+-   Current state of how diabetes is identified in Danish healthcare
+    registers.
+-   Challenges faced by researchers in this area, such as the limited
+    transparency in how diabetes is exactly classified in these sources
+    and how applying or using these approaches isn't very easy.
+-   How this algorithm and package contributes to discussions in this
+    space about how diabetes in classified in Danish register research
+    and how it is implemented.
+
+## Identifying type 1 and 2 diabetes cases in Danish healthcare registers
+
+### Danish register data infrastructure
+
+Many individual-level data (e.g. civil registration, public healthcare
+contacts, and drug prescriptions) are automatically collected on all
+residents in Denmark and stored in nationwide Danish registers by
+[Statistics Denmark](https://www.dst.dk/da/) and the [Danish Health Data
+Authority](https://sundhedsdatastyrelsen.dk/da). These agencies are
+legally allowed to give access to the register data for research
+purposes, which provides (authorized) researchers a set of common,
+extensive data sources to use for studies. Any researcher associated
+with an approved Danish research institute (mainly Danish universities)
+can apply for access, but fees and conditions apply.
+
+Register data is generally accessed and processed by approved
+researchers on remote servers operated by Statistics Denmark and the
+Danish Health Data Authority. The same raw data used by all researchers
+and the common virtual working environment potentially enables
+reproducible research. This means that any data processing workflow
+could be transferable and usable between research projects if the
+underlying code is designed with reproducibility in mind and the code is
+shared ("open-sourced") [@Marszalek2016]. While reproducibility in
+research relates to transparent reporting of methods to enable others to
+reproduce analyses and experiments, this also applies to a diabetes
+classification program, which - if reproducible - could be reused by any
+researcher with access to the necessary register data to dynamically
+identify a study population of individuals with diabetes for their
+research needs [@Dima2017].
+
+### Current Danish register-based diabetes classifiers
+
+In Denmark, the first resource readily available to researchers to
+identify diabetes cases through register data was the National Diabetes
+Register, which was established in 2006 [@Carstensen2011] but then
+discontinued in 2012.
+
+The next resource is the Register of Selected Chronic Diseases (RSCD),
```

Comment:
Add source for RSCD? 
The Danish Health Data Authority. The register of selected chronic diseases; 2014. Available from: https://www.esundhed.dk/Dokumentation/DocumentationExtended?id=29. Accessed November 25, 2022.